app/services/load_profiles/hierarchy.rb

```diff
@@ -0,0 +1,44 @@
+module LoadProfiles
+  class Hierarchy
+    #
+    # Query to sort Load Profile Categories hierarchly
+    #
+
+    def initialize
+    end
+
+    def tree_sort
+      with_paths.sort do |category_a, category_b|
+        category_a[:path] <=> category_b[:path]
```

Comment:
Maybe this could be...

```ruby
def tree_sort
  with_paths.sort do |category_a, category_b|
    sort_criteria(category_a) <=> sort_criteria(category_b)
  end
end

private

def sort_criteria(category)
  [category[:path], category[:load_profile_category].name.downcase]
end

```

... which would sort subcategories at the same level by their name?